go/processor/floria/processor.go

```diff
@@ -0,0 +1,115 @@
+
+package floria
+
+import (
+	"fmt"
+
+	"github.com/Fantom-foundation/Tosca/go/tosca"
+	"github.com/holiman/uint256"
+)
+
+func init() {
+	tosca.RegisterProcessorFactory("floria", newProcessor)
+}
+
+func newProcessor(interpreter tosca.Interpreter) tosca.Processor {
+	if interpreter == nil {
+		panic("interpreter is required")
+	}
+
+	return &processor{
+		interpreter: interpreter,
+	}
+}
+
+type processor struct {
+	interpreter tosca.Interpreter
+}
+
+func (p *processor) Run(
+	blockParams tosca.BlockParameters,
+	transaction tosca.Transaction,
+	context tosca.TransactionContext,
+) (tosca.Receipt, error) {
+
+	errorReceipt := tosca.Receipt{
+		Success: false,
+		GasUsed: transaction.GasLimit,
+	}
+
+	if err := buyGas(transaction, context); err != nil {
+		return errorReceipt, nil
+	}
+
+	if err := handleNonce(transaction, context); err != nil {
+		return errorReceipt, nil
+	}
+
+	if err := transferValue(transaction, context); err != nil {
+		return errorReceipt, nil
+	}
+
+	return tosca.Receipt{
+		Success:         true,
+		GasUsed:         transaction.GasLimit,
+		ContractAddress: nil,
+		Output:          []byte{},
+		Logs:            nil,
+	}, nil
+}
+
+func handleNonce(transaction tosca.Transaction, context tosca.TransactionContext) error {
+	stateNonce := context.GetNonce(transaction.Sender)
+	messageNonce := transaction.Nonce
+	if messageNonce != stateNonce {
+		return fmt.Errorf("nonce mismatch: %v != %v", messageNonce, stateNonce)
+	}
+
+	// Increment nonce
+	context.SetNonce(tosca.Address(transaction.Sender), context.GetNonce(tosca.Address(transaction.Sender))+1)
+
+	return nil
+}
+
+func buyGas(transaction tosca.Transaction, context tosca.TransactionContext) error {
+	gasPrice := transaction.GasPrice.ToUint256()
+	gas := uint256.NewInt(uint64(transaction.GasLimit))
+	gas = gas.Mul(gas, gasPrice)
+
+	// Buy gas
+	senderBalance := context.GetBalance(transaction.Sender).ToUint256()
+	if senderBalance.Cmp(gas) < 0 {
+		return fmt.Errorf("insufficient balance: %v < %v", senderBalance, gas)
+	}
+
+	senderBalance = senderBalance.Sub(senderBalance, gas)
+	context.SetBalance(transaction.Sender, tosca.ValueFromUint256(senderBalance))
+
+	return nil
+}
+
+func transferValue(transaction tosca.Transaction, context tosca.TransactionContext) error {
+	senderBalance := context.GetBalance(transaction.Sender).ToUint256()
+	transferValue := transaction.Value.ToUint256()
+	if senderBalance.Cmp(transferValue) < 0 {
+		return fmt.Errorf("insufficient balance: %v < %v", senderBalance, transferValue)
+	}
+
+	senderBalance = senderBalance.Sub(senderBalance, transferValue)
+	context.SetBalance(transaction.Sender, tosca.ValueFromUint256(senderBalance))
+
+	receiverBalance := context.GetBalance(*transaction.Recipient).ToUint256()
+	receiverBalance = receiverBalance.Add(receiverBalance, transferValue)
+	context.SetBalance(*transaction.Recipient, tosca.ValueFromUint256(receiverBalance))
```

Comment:
```suggestion
	senderBalance := context.GetBalance(transaction.Sender)
	transferValue := transaction.Value
	if senderBalance.Cmp(transferValue) < 0 {
		return fmt.Errorf("insufficient balance: %v < %v", senderBalance, transferValue)
	}

	senderBalance = senderBalance.Sub(senderBalance, transferValue)
	context.SetBalance(transaction.Sender, senderBalance)

	receiverBalance := context.GetBalance(*transaction.Recipient)
	receiverBalance = receiverBalance.Add(receiverBalance, transferValue)
	context.SetBalance(*transaction.Recipient, receiverBalance)
```
Maybe I'm missing something, but why are all the conversions to/from `U256` needed?

Reply:
I didn't consider that tosca.value also implements these functions. I also changed it in the other functions.